Spline path (#335)
rapyuta-robotics/RapyutaSimulationPlugins: docs/source/external_devices/character.rst

-----------

AI Character/Robot is an actor controlled via `BP_ROSAIController`. It has preset

> movements such as pick, drop, and navigation ando those movement can be triggered from both BP and ROS 2.

Reviewer: ```suggestion
movements such as pick, drop, and navigation and those movement can be triggered from both BP and ROS 2.
```

AI Controller
-------------

`RRAIRobotROSControllerParam <https://rapyutasimulationplugins.readthedocs.io/en/latest/doxygen_generated/html/d9/d59/class_u_r_r_a_i_robot_r_o_s_controller_param.html>`_
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

`RRAIRobotROSControllerParam <https://rapyutasimulationplugins.readthedocs.io/en/latest/doxygen_generated/html/d9/d59/class_u_r_r_a_i_robot_r_o_s_controller_param.html>`_
is a UActorComponent which has parameters for basic navigation functionality for RRAIRobotROSController.

> If controlled pawn has child class of RRAIRobotROSControllerPara, the param is passed to the controller when the pawn is possed.

Reviewer: ```suggestion
If controlled pawn has child class of RRAIRobotROSControllerParam, the param is passed to the controller when the pawn is possessed.
```

        2. LINEAR_MOVING: linear moving without AI
        3. ROTATING: Rotating without AI

BP_RRAIRobotROSControllerParam
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

BP_RRAIRobotROSControllerParam is child class of RRAIRobotROSControllerParam and has additional parameter such as BP_RRROS2AIControllerSplineParam.

BP_RRROS2AIControllerSplineParam
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> BP_RRROS2AIControllerSplineParam is a parameters for spline movement in BP_RRROSAIController.

Reviewer: ```suggestion
BP_RRROS2AIControllerSplineParam is a parameter for spline movement in BP_RRROSAIController.
```

.. list-table::
   :header-rows: 1

   * - Param Name
     - Type (Default)
     - Note
   * -
     -
     -
   * - Debug
     - bool (false)
     - Enables debug logging.
   * - Mode
     - int32 (0)
     - Defines movement mode:

>        0. OneTime: Move along spline one time adn stop at the end.

Reviewer: ```suggestion
       0. OneTime: Move along spline one time and stop at the end.
```

       1. LoopFromStart: Move along spline and navigate/teleport to start point when it reach end of spline and repeat movement.
       2. Reverse: Move along spline and reverse direction when it reach end/start of spline.
   * - FindClosestSpline
     - bool (false)
     - If true, it will find the closest BP_SplinePath from the current location.
   * - LookAheadDistance
     - float  (0.1)

>      - Distance to look ahead on the spline. Closest spline point from Current location + LookAheadDistance is used as target to mvoe.

Reviewer: ```suggestion
     - Distance to look ahead on the spline. Closest spline point from Current location + LookAheadDistance is used as target to move.
```
